### PR TITLE
Make multi-line text widget taller by default

### LINKED ISSUE #1197: Data capture - add support for the 'text' field type to render multiline text input
**Is your feature request related to a problem? Please describe.**
It's always frustrating when I need to verify a long line of text that has been written such as an address, in a single line text entry field. It means I have to scroll horizontally to verify that what I entered is correct.

**Describe the solution you'd like**
I would prefer if there was a multi line text entry field so I can verify the field at a glance without scrolling.

**Describe alternatives you've considered**
I've tried to use the 'text' type as specified in the FHIR documentation but this renders the same single line field.

**Additional context**
When using [this ](https://formdesigner.helsenorgelab.no/)form builder, it renders the 'string' and 'text' types differently as shown below.
146510573-255efcc9-3242-4a4c-96c6-073505630e83

However, when using the SDK, the 'string' and 'text' types are both rendered as single line fields as shown below.
146510602-ab76effb-9582-4b78-ba63-ef9a40fd5d0a

**Would you like to work on the issue?**
I don't know who can look into this but I would appreciate if someone looked into this as soon as possible.

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1197 

**Description**
Set a minimum number of lines for the multi-line edit text widget.

The widet will initially have 3 lines to indicate that the widget itself supports multi-line input. But as the user types the widget would expand vertically to accommodate more lines.

**Alternative(s) considered**
NA

**Type**
Bug fix

**Screenshots (if applicable)**

https://user-images.githubusercontent.com/28757340/181742578-fd3373ba-6774-4718-a24e-16ce501a0aaa.mp4

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextViewHolderFactory.kt [hunk 1]
```diff
@@ -122,4 +125,13 @@ abstract class QuestionnaireItemEditTextViewHolderDelegate(
...
     answer: QuestionnaireResponse.QuestionnaireResponseItemAnswerComponent?
   ): String
+
+  companion object {
+    /**
+     * The minimum number of lines to display for multi-line edit text widget. This gives the user
+     * the indication that the widget allows multi-line input. Without this, the widget will appear
+     * the same as a single-line edit text widget.
+     */
+    private const val MULTI_LINE_TEXT_EDIT_MIN_LINE_NUM = 3
```
Comment (aditya-07) [inline datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextViewHolderFactory.kt:135]: Would it make more sense to have it as a resource attribute with a default value 3 that can be overridden by the developer application as per their need?

**Review by santosh-pingle — APPROVED** [review]
Why do other 2 files shows code formatting changes?

LGTM, Thanks!

File: datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextViewHolderFactory.kt (hunk 1 above)
Comment (jingtang10) [reply datacapture/src/main/java/com/google/android/fhir/datacapture/views/QuestionnaireItemEditTextViewHolderFactory.kt]: done

**Comment by jingtang10** [comment]
> Why do other 2 files shows code formatting changes?
> 
> LGTM, Thanks!

i think probably other prs didn't run spotless properly? i'm not entirely sure

File: datacapture/src/main/res/layout/questionnaire_item_edit_text_multi_line_view.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        android:layout_height="wrap_content"
+    >
+
+        <com.google.android.material.textfield.TextInputEditText
+            android:id="@+id/text_input_edit_text"
+            style="?attr/questionnaireTextInputEditTextStyle"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:gravity="top"
+            android:minLines="3"
```
Comment (aditya-07) [inline datacapture/src/main/res/layout/questionnaire_item_edit_text_multi_line_view.xml:46]: Would this change allow the developer application to change the minLines ? Maybe minLines should be part of a new multi-line text style?

Comment (jingtang10) [reply datacapture/src/main/res/layout/questionnaire_item_edit_text_multi_line_view.xml:46]: thanks @aditya-07 - i think this should be probably done as a follow-up. We haven't heard any feature request about this one from the developers. Feel like this is a weaker requirement compared with what we still have to do to allow custom styles.

@santosh-pingle fyi @shelaghm fyi

**Comment by shelaghm** [comment]
@jingtang10 this looks good to me. It aligns with what I was thinking for paragraph text style. 
Attaching screenshot for reference below of mocks I did previously
1. One idea is to name this paragraph text input. Google forms uses this. What do you think? 
2. Do we want to add this as a separate component/widget or include in the existing text field component?
Screen Shot 2022-07-29 at 10 27 47
.

**Comment by jingtang10** [comment]
> @jingtang10 this looks good to me. It aligns with what I was thinking for paragraph text style. Attaching screenshot for reference below of mocks I did previously
> 
> 1. One idea is to name this paragraph text input. Google forms uses this. What do you think?

Do you mean change the widget's name to `Paragraph Text Input`? If we do that we should also make sure the single line text input is renamed to `Short answer text input` or something similar.

As an engineer, I sort of prefer the more functional names i.e. `single line` and `multi-line` because you may have answers that are not a paragraph (e.g. bullet points?) but still multi line? or answers that are quite long... but fits in a single line? so i think these names are potentially more accurate.

But I do understand why semantically it might be easier for people to understand what these means... so I'm ok with `short answer` and `paragraph` if that's the preferred wording from a ux perspective.

> 2. Do we want to add this as a separate component/widget or include in the existing text field component?

they are already separate components. but we're not showcasing them in the catalog app yet. i think the question here is wheather we put them in one single component or seprarate as two? both ok with me. single's probably fine and we can use the bottom sheet to toggle single/multi line?

> 
> Screen Shot 2022-07-29 at 10 27 47
> 
> .

**Review by aditya-07 — APPROVED** [review]
LGTM!

**Pull request merged**